### PR TITLE
 chore(build): upgrade to commons 0.10 

### PR DESCRIPTION
**Pull request by jeqo**
0.10.0 introduced an issue by shipping kafka-clients that is fixed on 0.10.1.
This was not spotted by the workflows as integration tests were disabled.

This PR enables integration tests on workflows, improves flaky integration tests, and upgrade commons.

### CODE REVIEW

File: .github/workflows/main_push_and_pull_request_workflow.yml [hunk 1]
```diff
@@ -10,17 +10,17 @@ jobs:
     strategy:
       matrix:
         java-version: [11, 17]
-        runs-on: [ubuntu-latest, macos-latest, windows-latest]
```
Comment (jeqo) [inline .github/workflows/main_push_and_pull_request_workflow.yml:13]: aligning with other Connectors, unless there's a reason to test against multiple OSs.

File: src/integration-test/java/io/aiven/kafka/connect/IntegrationTest.java [hunk 2]
```diff
@@ -314,8 +313,13 @@ final void oneFilePerRecordWithPlainValues(final String compression)
 
     @ParameterizedTest
     @ValueSource(strings = {"none", "gzip", "snappy", "zstd"})
-    final void groupByKey(final String compression) throws ExecutionException, InterruptedException, IOException {
-        final Map<String, String> connectorConfig = awsSpecificConfig(basicConnectorConfig(CONNECTOR_NAME));
+    void groupByKey(final String compression, final TestInfo testInfo)
+        throws ExecutionException, InterruptedException, IOException {
+        final var topicName0 = IntegrationBase.topicName(testInfo);
+        final var topicName1 = IntegrationBase.topicName(testInfo) + "_1";
```
Comment (HelenMel) [inline src/integration-test/java/io/aiven/kafka/connect/IntegrationTest.java:319]: Oh, here is how you overcome a limitation I mentioned :+1:

Comment (HelenMel) [reply src/integration-test/java/io/aiven/kafka/connect/IntegrationTest.java:319]: [Super minor] In the future, we could make it more generic to create all required topics in advance - considering also reusing the Method name for this purpose.

**Review by HelenMel — APPROVED** [review]
A few minor comments

**Comment by HelenMel** [comment]
LGTM, lets merge!

**Pull request merged**